File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
//bfs traversal e queue ta empty hobar age joto bar cholche setai holo tar max level ba height
//first e 3 ta queue te elo to cnt=1
//trpr 9 20 ta queue te elo to cnt=2
//trpr 15 7 ta queue te elo to cnt=3
    int maxDepth(TreeNode* root) {
        if(root==NULL){
            return 0;
        }
        int cnt=0;
        queue<TreeNode*>q;
        q.push(root);
        while(!q.empty()){
          int n=q.size();
          for(int i=0;i<n;i++){
            TreeNode* node=q.front();
            q.pop();
            if(node->left!=NULL){
                q.push(node->left);
            }
            if(node->right!=NULL){
                q.push(node->right);
            }
          }
          cnt++;
        }
        return cnt;
    }
};

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    //recursive function to calculate height of the tree
    int f(TreeNode root){
        //base case
        if(root==null){
            return 0;
        }
        //ekta node er height holo tar left subtree and right subtree er moddhe maximum depth jar , ar oi node ta concider korar jonno 1 plus korte 
        //hobe
        int left_height=f(root.left);
        int right_height=f(root.right);
        //eta holo last theke oi node obdhi current height
        int height=Math.max(left_height,right_height)+1;
        return height;
    }
    public int maxDepth(TreeNode root) {
        int ans=f(root);
        return ans;
    }
}